firmware/library/L1_Drivers/system_clock.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+
+class SystemClock
+{
+ public:    
+      virtual void selectClockSource(bool option) = 0;
+      virtual void setMainPLL() = 0;
+      virtual void Init() = 0; 
+      virtual void setIRCMode() = 0;
+      virtual void setMainOscillatorMode() = 0;
+      virtual void setRTCMode() = 0;
+      virtual void setSpeed(uint8_t speed);
+      virtual bool isMain() = 0;
```

Comment:
If I wanted to know which clock source this was on, I would have to check all three methods. How about having 1 method that returns a `enum` class for the mode that its on. You could use the enum for setting modes as well.